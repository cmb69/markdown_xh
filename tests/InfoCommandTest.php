<?php

namespace Markdown;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeSystemChecker;
use Plib\SystemChecker;
use Plib\View;

class InfoCommandTest extends TestCase
{
    private SystemChecker $systemChecker;
    private View $view;

    public function setUp(): void
    {
        $this->systemChecker = new FakeSystemChecker();
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["markdown"]);
    }

    private function sut(): InfoCommand
    {
        return new InfoCommand("./", $this->systemChecker, $this->view);
    }

    public function testRendersPluginInfo(): void
    {
        $response = $this->sut()();
        $this->assertSame("Markdown 0.1-dev", $response->title());
        Approvals::verifyHtml($response->output());
    }
}
