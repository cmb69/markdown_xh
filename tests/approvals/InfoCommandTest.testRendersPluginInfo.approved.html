
<article class="markdown_plugininfo">
  <h1>Markdown 0.1-dev</h1>
  <section class="markdown_syscheck">
    <h2>System Check<h2>
    <p class="xh_fail">PHP version ≥ 7.4.0: bad</p>
    <p class="xh_fail">PHP extension DOM available: bad</p>
    <p class="xh_fail">CMSimple_XH version ≥ 1.7.0: bad</p>
    <p class="xh_fail">Plib_XH version ≥ 1.10: bad</p>
    <p class="xh_fail">./css/ is writable: bad</p>
    <p class="xh_fail">./languages/ is writable: bad</p>
  </section>
</article>
